File: src/main/java/net/vidageek/games/task/IndexedTask.java
package net.vidageek.games.task;

public final class IndexedTask implements Task {

	private final Task delegate;
	private final int index;

	public IndexedTask(final Task delegate, final int index) {
		this.delegate = delegate;
		this.index = index;
	}

	public int getIndex() {
		return index;
	}

	public JudgedTask judge(final String challenge) {
		return delegate.judge(challenge);
	}

	public String getDescription() {
		return delegate.getDescription();
	}

	public String getChallenge() {
		return delegate.getChallenge();
	}

}

File: games/game/src/main/scala/vggames/shared/task/IndexedTask.scala
package vggames.shared.task;

import vggames.shared.task.status.Error
import scala.util.Try

class IndexedTask(val group : TaskGroup, delegate : Task, val index : Int) {

  def judge(challenge : String) : JudgedTask = {
    Try(delegate.judge(challenge)).
      recover { case e : Exception => new Error(e) }.
      get
  }

  def resource = delegate.resource

  def challenge : String = delegate.challenge

  def groupName = group.htmlName

  def groupCode = group.groupName

  override def toString : String = delegate.toString

  def extraData = delegate.extraData

}
